### ISSUE TITLE
Linter:toggle and Linter:lint need to be worked on

### ISSUE THREAD

**Issue by posita** [atom-community/linter#748]
This is possibly related to #716. To reproduce:

1. Open a linter-enabled buffer in a new tab (e.g., `foo.py` below with `linter-pylint`)
1. Verify that linter warnings are displayed
1. Select "Linter:Toggle" from the command palette
1. Note that the linter warnings have disappeared
1. Select "Linter:Toggle" again from the command palette

Expected results: the warnings reappear

Actual results: the warnings stay hidden

Other observations:

1. Navigate away from the tab without closing it (i.e., activate a different tab in the same window)
1. Navigate back to the toggled tab
1. Note that warnings are *still* not displayed
1. Close the tab and reopen it (e.g., `cmd-shift-t` or `ctrl-shift-t`)
1. Note that warnings are now displayed

```py
# foo.py
import foo # this package doesn't exist and should give a warning
```

**Comment by Eschon** [comment]
It seems that linter doesn't automatically lint when turning back on. If you select `Linter: Lint` from the command palette the warnings reappear.

**Comment by steelbrain** [comment]
Can you reproduce this after checking out #738

**Comment by posita** [comment]
> Can you reproduce this after checking out #738
@steelbrain, I'd be happy to, but I'm not building/installing from source (I'm not an Atom or Atom plugin developer). I can think of a couple options:

1. I can wait for a release with that PR merged and test then; or
1. If you can point me to a quick-and-dirty way to install (and uninstall) a plugin from source, I can try that.

**Comment by steelbrain** [comment]
@posita You don't actually need to build anything, here's how you can do it
```
cd ~/.atom/packages
mv linter /tmp/linter-old
git clone https://github.com/atom-community/linter --deph=1
cd linter
npm install
```

**Comment by posita** [comment]
@steelbrain, you mean `apm`, not `npm`, correct?

**Comment by steelbrain** [comment]
For `linter`, both should work fine as we aren't using any native modules.

**Comment by posita** [comment]
Okay, with `master`, `Linter:Toggle` doesn't seem to have *any* effect (it doesn't turn off warnings). At least not with `linter-pylint`.

Here's what I did:

```sh
% echo 'import unusedpkgthatdoesnotexist' >bar.py
% atom bar.py
```

![screen shot 2015-07-29 at 11 16 33](https://cloud.githubusercontent.com/assets/222581/8965998/aa879808-35e3-11e5-8912-d11f944b7922.png)

Then I selected `Linter:Toggle` from the command palette.

![screen shot 2015-07-29 at 11 16 53](https://cloud.githubusercontent.com/assets/222581/8965999/aa88162a-35e3-11e5-98af-348e932a3b65.png)

Nothing changed (warnings were stilled displayed).

![screen shot 2015-07-29 at 11 16 33](https://cloud.githubusercontent.com/assets/222581/8965998/aa879808-35e3-11e5-8912-d11f944b7922.png)

I tried this several times, all with the same effect.

**Comment by steelbrain** [comment]
Right, `linter:lint` __might__ be broken in master, I'll try to fix it when I have some time.

**Comment by posita** [comment]
Should I try steelbrain/port-editor-registry@74bacf8 then?

**Comment by steelbrain** [comment]
There is a possibility that it'll fix your problem, but I am not entirely sure as I didn't address it there.

**Comment by posita** [comment]
@74bacf8 exhibits the same no-effect behavior described in my https://github.com/atom-community/linter/issues/748#issuecomment-126048390.

**Comment by steelbrain** [comment]
Fixed by https://github.com/atom-community/linter/pull/791
